L1Trigger/L1THGCal/src/HGCalCoarseTriggerCellMapping.cc

```diff
@@ -0,0 +1,301 @@
+#include "L1Trigger/L1THGCal/interface/HGCalCoarseTriggerCellMapping.h"
+#include "DataFormats/ForwardDetId/interface/HGCalTriggerDetId.h"
+
+
+HGCalCoarseTriggerCellMapping::
+HGCalCoarseTriggerCellMapping(const std::vector<unsigned>& ctcSize )
+  :  ctcSize_(!ctcSize.empty() ? ctcSize : 
+	      std::vector<unsigned>{2,2,2,2})
+{
+      if ( ctcSize_.size() != kNLayers_ ){
+        throw cms::Exception("HGCTriggerParameterError")
+            << "Inconsistent size of coarse trigger cell size vector" << ctcSize_.size() ;
+    }
+
+    for(auto ctc : ctcSize_) checkSizeValidity(ctc);
+
+}
+
+
+const std::map<int,int> 
+HGCalCoarseTriggerCellMapping::kSplit_ = {
+  {kCTCsizeIndividual_, kSplit_v8_Individual_},
+  {kCTCsizeVeryFine_, kSplit_v8_VeryFine_},
+  {kCTCsizeFine_, kSplit_v8_Fine_},
+  {kCTCsizeMid_, kSplit_v8_Mid_},
+  {kCTCsizeCoarse_, kSplit_v8_Coarse_}
+};
+
+const std::map<int,int> 
+HGCalCoarseTriggerCellMapping::kSplit_v9_ = {
+  {kCTCsizeIndividual_, kSplit_v9_Individual_},
+  {kCTCsizeVeryFine_, kSplit_v9_VeryFine_},
+  {kCTCsizeFine_, kSplit_v9_Fine_},
+  {kCTCsizeMid_, kSplit_v9_Mid_},
+  {kCTCsizeCoarse_, kSplit_v9_Coarse_},
+};
+
+const std::map<int,int> 
+HGCalCoarseTriggerCellMapping::kSplit_v9_Scin_ = {
+  {kCTCsizeIndividual_, kSplit_v9_Scin_Individual_},
+  {kCTCsizeVeryFine_, kSplit_v9_Scin_VeryFine_},
+  {kCTCsizeFine_, kSplit_v9_Scin_Fine_},
+  {kCTCsizeMid_, kSplit_v9_Scin_Mid_},
+  {kCTCsizeCoarse_, kSplit_v9_Scin_Coarse_},
+};
+
+void
+HGCalCoarseTriggerCellMapping::checkSizeValidity(int ctcSize )const{
+  if ( ctcSize!=kCTCsizeFine_ && ctcSize!=kCTCsizeCoarse_ && ctcSize!=kCTCsizeMid_ && ctcSize!=kCTCsizeVeryFine_ && ctcSize!=kCTCsizeIndividual_){
+    throw cms::Exception("HGCTriggerParameterError")
+      << "Coarse Trigger Cell should be of size "<<
+      kCTCsizeIndividual_ << " or " << kCTCsizeVeryFine_ << " or " << kCTCsizeFine_ << 
+      " or " << kCTCsizeMid_ << " or " << kCTCsizeCoarse_;
+  }
+}
+
+uint32_t
+HGCalCoarseTriggerCellMapping:: getEvenDetId(uint32_t tcid) const {
+  
+  uint32_t evenid = 0;
+  DetId tc_Id( tcid );
+  if ( tc_Id.det() == DetId::Forward ){//V8
+    if( triggerTools_.isScintillator(tcid) ){
+      evenid = tcid; //stc not available in scintillator for v8
```

Comment:
If it would be a useful feature I can add it - let me know.

Reply:
It depends on the amount of work expected to implement this feature.

Reply:
I think the amount of work would be small if I clearly knew the cell numbering convention. It would then just be a case of finding the right masks.

Reply:
Unfortunately there is no convention in that case because we map V9 detids to V8 detids. You can maybe have a quick look at the geometry ntuple in that case, and decide whether it will be easy or not.

Reply:
![scintillatorv8](https://user-images.githubusercontent.com/43339682/58492733-6ce4e480-8169-11e9-9fe1-24b8a3b60510.png)
At far as I can tell, the cells in each wafer are numbered 1-140, in a 10x14 grid.
So creating coarse cells of size 2 would be easy, but 4, 8, 16 become more difficult (unless you have a good idea).

Reply:
Ok, thanks. So let's forget about this. Anyway we'll have to switch to `V9Imp2` at some point